src/CistromeHGWConsumer.js

```diff
@@ -0,0 +1,186 @@
+     * Updates the mapping of `horizontal-multivec` track IDs 
+     * to sibling `viewport-projection-horizontal` track IDs.
+     */
+    const onViewConfig = useCallback((newViewConfig) => {
+        const newTrackIds = getHMTrackIdsFromViewConfig(newViewConfig);
+        const newSiblingTrackIds = {};
+        for(let trackId of newTrackIds) {
+            // Each trackId is actually an object `{ viewId, trackId }`, which is why we want trackId.trackId.
+            newSiblingTrackIds[trackId.trackId] = getSiblingVPHTrackIdsFromViewConfig(newViewConfig, trackId.trackId);
+
+            const newSelectedRows = getHMSelectedRowsFromViewConfig(newViewConfig, trackId.viewId, trackId.trackId);
+            if(
+                !context.state[trackId.viewId] 
+                || !context.state[trackId.viewId][trackId.trackId] 
+                || !isEqual(newSelectedRows, context.state[trackId.viewId][trackId.trackId].selectedRows)
+            ) {
+                context.dispatch({
+                    type: ACTION.SELECT_ROWS,
+                    viewId: trackId.viewId,
+                    trackId: trackId.trackId,
+                    selectedRows: newSelectedRows
+                });
+            }
+
+            // TODO: dispatch for highlighting
+        }
+        setTrackIds(newTrackIds);
+        setSiblingTrackIds(newSiblingTrackIds);
+    }, []);
+
+    // Function to get a track object from the higlass API.
+    const getTrackObject = useCallback((viewId, trackId) => {
+        try {
+            return hgRef.current.api.getTrackObject(viewId, trackId);
+        } catch(e) {
+            return null;
+        }
+    }, []);
+
+    const setTrackSelectedRows = useCallback((viewId, trackId, selectedRows) => {
+        const currViewConfig = hgRef.current.api.getViewConfig();
+        const newViewConfig = updateViewConfigOnSelectRowsByTrack(currViewConfig, selectedRows, viewId, trackId);
+        hgRef.current.api.setViewConfig(newViewConfig).then(() => {
+            onViewConfig(newViewConfig);
+        });
+    }, [hgRef]);
+    
+
+    // Function for child components to call to "register" their draw functions.
+    const drawRegister = useCallback((key, drawFunction) => {
+        drawRef.current[key] = drawFunction;
+    }, [drawRef]);
+
+    // Process the options prop (initially, upon sorting, etc).
+    useEffect(() => {
+        setOptions(processWrapperOptions(optionsRaw));
+
+        const sortToken = PubSub.subscribe(EVENT.SORT, (msg, data) => {
+            const newOptionsRaw = updateRowSortOptions(optionsRaw, data);
+            const newOptions = processWrapperOptions(newOptionsRaw)
+            setOptions(newOptions);
+
+            const trackOptions = getTrackWrapperOptions(newOptions, data.viewId, data.trackId);
+            const newSelectedRows = selectRows(context.state[data.viewId][data.trackId].rowInfo, trackOptions);
+            setTrackSelectedRows(data.viewId, data.trackId, newSelectedRows);
```

Comment:
(Let's discuss this later in the lab.)

When a user press a sort button, should we (1) add the options to `globalDefaults` and rearrange the entire multivec tracks? (2) Or should we add the options to that of a certain track and rearrange the single track? Currently, it seems the options are added to globalDefaults (part of 1) but a single track is rearranged (part of 2).